src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -31,7 +32,27 @@ public class CompareSymbolsCorrectlyAnalyzer : DiagnosticAnalyzer
             description: s_localizableDescription,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+        public static readonly DiagnosticDescriptor GetHashCodeRule = new DiagnosticDescriptor(
+            DiagnosticIds.CompareSymbolsCorrectlyRuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: s_localizableDescriptionGetHashCode,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public static readonly DiagnosticDescriptor CollectionRule = new DiagnosticDescriptor(
```

Comment:
There is currently no difference between this descriptor and the `EqualityDescriptor`, the only goal is to prevent registering the code-fix for the wrong case.